src/content/en/fundamentals/instant-and-offline/get-started/textcontent-3.md

```diff
@@ -0,0 +1,322 @@
+
+and Gulp,
+
+[https://www.npmjs.com/package/gulp-html-minifier](https://www.npmjs.com/package/gulp-html-minifier)
+
+and a popular package 
+
+[https://www.npmjs.com/package/html-minifier](https://www.npmjs.com/package/html-minifier)
+
+that includes a useful chart comparing its compression results with other methods.
+
+### Frameworks
+
+Of course, chances are good that you do (or will) use a framework, IDE, or other structured 
+environment as you write, rather than copy/pasting your code into a web app one file at a time. 
+Most modern systems have built-in facilities for keeping development files separate from 
+deployment during the build process, and are able to perform a variety of transformations, 
+such as minification, along the way..
+
+For example, a Gulp development-to-deployment task that includes HTML minification might 
+look like this.
+
+```
+var gulp = require('gulp');
+
+var htmlmin = require('gulp-html-minifier');
+
+ 
+
+gulp.task('minify', function() {
+
+  gulp.src('./src/*.html') //development location
+
+    .pipe(htmlmin({collapseWhitespace: true}))
+
+    .pipe(gulp.dest('./dist')) //deployment location
+
+});
+```
+
+The execution command for that task would then be:
+
+`gulp minify`
+
+(Source: [npmjs](https://www.npmjs.com/package/gulp-html-minifier))
+
+Note that the task pipes the HTML files from their original location to the deployment 
+location, minifying them during the build. This not only prevents potentially irreversible 
+modification of the source files, but avoids contaminating the production environment during 
+later development and testing.
+
+## Compress Text Resources
+
+So far we've talked about compression in terms of individual image and text files. But it 
+would also be helpful if we could get our server to automatically compress entire file sets 
+as well, and that's where Gzip comes in.
+
+Gzip is an application (and its file format) for compressing and decompressing files. Like 
+solo file compression, it reduces the time required to deliver a server response by reducing 
+the size of the resource. It is available via the GNU Project site.
+
+[https://www.gnu.org/software/gzip/](https://www.gnu.org/software/gzip/)
+
+Gzip performs best on text resources, and can regularly achieve up to 70% compression (even 
+higher for large files). However, Gzipping non-text resources, such as images, that have 
+already been individually compressed generally provides no significant size reduction.
+
+Unlike desktop or browser-based local compression, Gzip works at the server to identify and 
+process certain file types that you specify. While all modern browsers support Gzip compression 
+for HTTP requests, you must properly configure your server to deliver the compressed resource 
+when it is requested. Different server types, of course, have different setup requirements. 
+You configure the Apache server, for example -- which nearly half of all active websites 
```

Comment:
Suggestion to not refer to stats on servers. Also, just give one example that matches your sample code.